enterprise/server/remote_execution/containers/ociruntime/ociruntime_test.go

```diff
@@ -922,6 +922,41 @@ func TestOverlayfsEdgeCases(t *testing.T) {
 	assert.Equal(t, 0, res.ExitCode)
 }
 
+func TestOverlayfsHighLayerCount(t *testing.T) {
+	setupNetworking(t)
+
+	image := "ghcr.io/avdv/nix-build@sha256:5f731adacf7290352fed6c1960dfb56ec3fdb31a376d0f2170961fbc96944d50"
```

Comment:
The layering logic is starting to get complicated - it's kind of hard to understand the layer ordering now that the "groups" have to be in reverse-manifest-order and so do the individual layers within those merged groups.

So I feel like it'd be better to have a more explicit test on how we're handling the layers, rather than an integration test. For example, create a fake slice of 30 `image.Layer` structs, pass that to a func like `GetOverlayMounts()` which returns some data needed to create the overlays, then assert that it returns 2 dirs, one with 10 layers and another with 20 layers (and make sure the order of the layers is correct). The "20" const could also be a parameter and it could be set to 3 to make the test easier to follow.

Doing a more specialized test like this would also let us avoid depending on this heavyweight image to verify correctness

Reply:
Yeah with the latest version of this PR, I do think that the logic is a bit convoluted (multiple slices.Reverse calls) and could use some tests.

I will look for a way to export part of `createRootfs()` logic for testing.

Reply:
I gave this a try and when I tried to do the refactoring, making the test inspecting the directory and mount structure felt... wrong? It felt like we were testing the implementation details, which would lock us out of changing it up in the future.

Instead, I think we should just test the higher-level effect which matters to our users.
So I added a test that creates X layers on top of busybox container image.
Each layer includes an `a.txt` file with a different value and we simply assert that the last layer wins by running `cat /a.txt`.

Let me know what you think.